src/test/java/org/springframework/samples/petclinic/performance/N1OptimizedRestControllerTest.java

```diff
@@ -0,0 +1,302 @@
+        assertThat(owners.length).isGreaterThan(0);
+
+        // Get query metrics
+        SqlQueryInterceptor.QueryMetrics metrics = sqlQueryInterceptor.getMetrics();
+
+        // Count owners with pets
+        int ownersWithPets = 0;
+        int totalPets = 0;
+        for (OwnerDto owner : owners) {
+            if (owner.getPets() != null && !owner.getPets().isEmpty()) {
+                ownersWithPets++;
+                totalPets += owner.getPets().size();
+            }
+        }
+
+        // Log detailed analysis
+        logger.info("=".repeat(80));
+        logger.info("OWNER->PET N+1 ANALYSIS: GET /api/owners");
+        logger.info("=".repeat(80));
+        logger.info("Number of owners loaded: {}", owners.length);
+        logger.info("Owners with pets: {}", ownersWithPets);
+        logger.info("Total pets loaded: {}", totalPets);
+        logger.info("Total SQL queries executed: {}", metrics.getQueryCount());
+        logger.info("SELECT queries: {}", metrics.getQueryTypeAnalysis().getSelectCount());
+        logger.info("JOIN queries: {}", metrics.getQueryTypeAnalysis().getJoinCount());
+        logger.info("Execution time: {}ms", metrics.getExecutionTimeMs());
+        logger.info("Likely N+1 problem: {}", metrics.getQueryTypeAnalysis().isLikelyN1Problem());
+        logger.info("-".repeat(80));
+        logger.info("SQL Queries executed:");
+        for (int i = 0; i < metrics.getQueries().size(); i++) {
+            logger.info("{}. {}", (i + 1), metrics.getQueries().get(i));
+        }
+        logger.info("=".repeat(80));
+        logger.info("N+1 ANALYSIS RESULT:");
+        logger.info("Expected for optimal: 1 query with JOINs (owners + pets)");
+        logger.info("Expected for N+1 problem: 1 + N queries (1 for owners + {} for pets)", ownersWithPets);
+        logger.info("Actual queries executed: {}", metrics.getQueryCount());
+
+        if (metrics.getQueryCount() == 1) {
+            logger.info("✅ EXCELLENT: No N+1 problem! Single query with JOINs.");
+        } else if (metrics.getQueryCount() <= 2) {
+            logger.info("✅ GOOD: Minimal queries, likely optimized.");
+        } else if (metrics.getQueryTypeAnalysis().isLikelyN1Problem()) {
+            logger.warn("❌ N+1 PROBLEM DETECTED: {} queries for {} owners with pets",
+                metrics.getQueryCount(), ownersWithPets);
+        } else {
+            logger.info("⚠️  MODERATE: {} queries executed, needs investigation", metrics.getQueryCount());
+        }
+        logger.info("=".repeat(80));
+
+        // Verify that owners with pets have them populated
+        for (OwnerDto owner : owners) {
+            if (owner.getPets() != null) {
+                logger.info("Owner {} {} has {} pets",
+                    owner.getFirstName(), owner.getLastName(), owner.getPets().size());
+            }
+        }
+
+        // Log conclusion
+        if (metrics.getQueryTypeAnalysis().isLikelyN1Problem()) {
```

Comment:
I'm a bit confused why do we say that n+1 is likely (`selectCount > 5 && joinCount < (selectCount / 3);`), but then do no assert this. Lets simplify.

The quieries list is super useful, but this analysis with ifs looks confusing.